ramips: add support for Zorlik ZL5900V2

vianney opened the pull request:
The Zorlik ZL5900V2 is an unbranded clone of HAME MPR-A1/2.  It is
marketed as "3G Wi-Fi Router".  Only the PCB has the model name
"ZL5900V2" printed on it.

Specifications:
- Ralink RT5350F (360 MHz)
- 32 MB RAM
- 8 MB Flash
- 802.11bgn 1T1R
- 1x 10/100 Mbps Ethernet
- 1x USB 2.0 (Type-A)
- 5200 mAh battery

The ramdisk image (not the squashfs sysupgrade image) can be flashed
through the web interface (named "GoAhead") of the factory firmware.
However, as the factory firmware does not cleanly unmount the rootfs
before flashing, the device may hang instead of rebooting after
successful write.  Power cycling the device gets you in OpenWrt where
the squashfs image may be flashed through normal sysupgrade procedure.

Signed-off-by: Vianney le Clément de Saint-Marcq <code@quartic.eu>

vianney committed:
ramips: add support for Zorlik ZL5900V2

The Zorlik ZL5900V2 is an unbranded clone of HAME MPR-A1/2.  It is
marketed as "3G Wi-Fi Router".  Only the PCB has the model name
"ZL5900V2" printed on it.

Specifications:
- Ralink RT5350F (360 MHz)
- 32 MB RAM
- 8 MB Flash
- 802.11bgn 1T1R
- 1x 10/100 Mbps Ethernet
- 1x USB 2.0 (Type-A)
- 5200 mAh battery

The ramdisk image (not the squashfs sysupgrade image) can be flashed
through the web interface (named "GoAhead") of the factory firmware.
However, as the factory firmware does not cleanly unmount the rootfs
before flashing, the device may hang instead of rebooting after
successful write.  Power cycling the device gets you in OpenWrt where
the squashfs image may be flashed through normal sysupgrade procedure.

Signed-off-by: Vianney le Clément de Saint-Marcq <code@quartic.eu>

mkresin commented on target/linux/ramips/dts/ZL5900V2.dts:
```diff
@@ -0,0 +1,99 @@
+&pinctrl {
+	state_default: pinctrl0 {
+		gpio {
+			ralink,group = "i2c", "jtag", "uartf", "led";
```
only `ralink,group = "jtag", "uartf";` should be required here. No pin of of the i2c or led group is used,

vianney replied:
Fixed. I also removed the gpio1 node, because GPIOs > 22 are unused.

mkresin commented on target/linux/ramips/dts/ZL5900V2.dts:
```diff
@@ -0,0 +1,99 @@
+		compatible = "gpio-leds";
+
+		lan {
+			label = "zl5900v2:green:lan";
```
Wouldn't it make sense to use this LED for the lan port status? Add a section for your board to `target/linux/ramips/base-files/etc/board.d/01_leds` with

```
zorlik,zl5900v2)
	ucidef_set_led_netdev "lan" "lan" "$boardname:green:lan" eth0
	;;
```

vianney replied:
Done.

mkresin reviewed (changes requested):
Excellent job so far. Nevertheless I found some minor issues.

Please use ```git commit --amend``` to add changes to your commit. Use ```git push --force``` to overwrite your github repository/branch. The PR will afterwards update automatically. Please don't close this PR and open a new one.

vianney commented:
@mkresin Thanks for the code review.  I have updated my patch and rebased it on master.  The downside is that USB does not work anymore on 4.14.  I will make a separate PR to fix that.

mkresin commented:
merged, thanks!
